### PR TITLE
Allow for numeric components in macro define paths

### PR DESCRIPTION
**Pull request by amylizzle**
Allows macros to define paths like `/obj/thing_3/`

### CODE REVIEW

File: DMCompiler/Compiler/DM/DMLexer.cs [hunk 1]
```diff
@@ -30,6 +30,11 @@ public sealed class DMLexer : TokenLexer {
             "..."
         };
 
+        private static readonly List<TokenType> ValidIdentifierComponents = new(2) {
+            TokenType.DM_Preproc_Identifier,
+            TokenType.DM_Preproc_Number
```
Comment (ike709) [inline DMCompiler/Compiler/DM/DMLexer.cs:35]: Does checking 2 types in 1 spot really justify moving this to a list?

Comment (amylizzle) [reply DMCompiler/Compiler/DM/DMLexer.cs:35]: Not really, but I was thinking this would be a much better place to handle the "operator[]" nonsense that I did kinda hackily in the operator overloads PR.

Which is to say, I intend to add a bunch more things to this list.

File: Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/obj/thing_1/dodaa
+	name = @#underscore 1 test#
```
Comment (wixoaGit) [inline Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm:2]: What does this have to do with the test?

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm (hunk 2 above)
Comment (amylizzle) [reply Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm:2]: nothing, the erroring code was using a rawstring, and I didn't see a need to remove it.

Comment (wixoaGit) [reply Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm:2]: This should be removed to keep the test focused on the macro problem. If you want to test raw strings, a new test for it could be created under `Tests/Text/`.

Comment (amylizzle) [reply Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm:2]: Sure thing, I'll do it now

Comment (wixoaGit) [reply Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm]: I meant the `/obj/thing_1/dodaa` in general

Comment (amylizzle) [reply Content.Tests/DMProject/Tests/Preprocessor/macro_numeric_path.dm]: oh, the error was occurring when a path was defined as `/obj/thing_##NUM` where `##NUM` was numeric. It didn't error on fully defined paths like `/obj/thing_1`. I was just making sure I didn't alter that behaviour.

**Review by wixoaGit — APPROVED** [review]
